lib/superstring/index.js

```diff
@@ -64,6 +64,46 @@ _.assign(SuperString.prototype, /** @lends SuperString.prototype */ {
         return this;
     },
 
+    async replaceAsync (regex, fn) {
+        var replacements = 0; // maintain a count of tokens replaced
+
+        // to ensure we do not perform needless operations in the replacement, we use multiple replacement functions
+        // after validating the parameters
+        const replacerFn = _.isFunction(fn) ?
+            function () {
+                replacements += 1;
+
+                return fn.apply(this, arguments);
+            } :
+            // this case is returned when replacer is not a function (ensures we do not need to check it)
+            /* istanbul ignore next */
+            function () {
+                replacements += 1;
+
+                return fn;
+            };
+
+        let index = 0,
+            match;
+
+        while ((match = regex.exec(this.value.slice(index)))) {
+            try {
+                // eslint-disable-next-line no-await-in-loop
+                let value = await replacerFn(...match);
+
+                index += match.index;
+                this.value = this.value.slice(0, index) + value + this.value.slice(index + match[0].length);
+                index += match[0].length;
+            }
+            catch (_err) { /* empty */ }
+        }
```

Comment:
custom string replace logic instead of using String.replace ( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace ) as it does not support async replacer fn